src/state/slices/columnsSlice.tsx

```diff
@@ -0,0 +1,160 @@
+import { createSelector, createSlice } from '@reduxjs/toolkit';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { Column } from 'react-table';
+import { DropResult } from 'react-beautiful-dnd';
+import { RootState } from '../store';
+import { ColumnState, Order } from '../../app.types';
+import { resultsPerPage } from '../../recordGeneration';
+
+// Define a type for the slice state
+interface ColumnsState {
+  columnStates: {
+    [id: string]: ColumnState;
```

Comment:
I feel like there's potential for confusion here. `ColumnState` and `ColumnsState`. Could we change the latter to `TableState`? I'm also wary of getting in the way of React Table's native `state` object, in case we need this more down the line in `table.component.tsx` but I still feel this needs a name change

Reply:
Ah true - the convention is for the `State` interface for a slice to share the same name - hence the name as it's the `Columns` slice. I only named this way after the Epic name - this slice could easily be renamed `TableSlice` and thus `ColumnsState` -> `TableState`, `ColumnsReducer` -> `TableReducer`, `state.columns` -> `state.table` etc.

The alternative is to refactor `columnStates`/`ColumnState` - the idea of this is to store all the client-side info of a column - e.g. the word wrap state, potentially the column sizes. Also, we could also refactor the `sort` state to live in there as well. Or, they could all be separate parts of the redux state, so `wordWrap` refactored to look more like the current `sort` state.

Reply:
I like the idea of Column and Table respectively. As for where sort can live, I think it's all about where makes sense and is easily accessible. Ultimately, it's a parameter passed with several others to the API query, unlike the word wrap which is a client-side piece of info. So maybe it makes sense for those to be two separate entities - TableState for how the table is being sorted as a whole and ColumnState for how the user has chosen to manipulate columns client-side

Reply:
So are you proposing we do then?
- Rename `ColumnsSlice` to `TableSlice` (and other associated terms in the slice) and keep `columnState` with it's current name?
- Separate out the `columnState` state into it's own slice called `ColumnsSlice`, and rename the existing slice to `TableSlice` as that's more for data passed to the backend? Where would the `selectedIds` state go then? As that presumably will eventually be used to selectively query for data right?
- A third option I didn't understand 😅

Reply:
Apologies, the first option gets my vote